File: Padrões de Projeto/interator/interator/src/classes/BilheteLoteria.java
/*
 *
 */

package classes;

import java.util.ArrayList;
import java.util.Iterator;

/**
 *
 * @author Laranja
 */

public class BilheteLoteria implements Iterator<Integer> {

	private ArrayList<Integer> numerosBilhete;
	private int posicao = 0;

	public BilheteLoteria(ArrayList<Integer> numerosBilhete) {
		this.numerosBilhete = numerosBilhete;
	}

	public Integer next() {
            int numero = numerosBilhete.get(posicao);
            posicao++;
            return numero;
	}

	public boolean hasNext() {
            if (posicao >= numerosBilhete.size() || numerosBilhete.get(posicao) == null) {
                return false;
            }
            return true;
	}

}

File: C#/genericEinterface/classes/BilheteLoteria.cs
using System.Collections.Generic;
using genericEinterface.interfaces;

namespace genericEinterface.classes
{
    public class BilheteLoteria : IBilheteLoteria<int>
    {
        public List<int>numerosBilhete {get; set;}
        private int posicao;

        public BilheteLoteria(List<int> numerosBilhete)
        {
            this.numerosBilhete = numerosBilhete;
        }

        public int next() => numerosBilhete[posicao++];
        
        public bool hasNext() => (!(posicao >= numerosBilhete.Count || numerosBilhete[posicao] == null));
    }
}